### DISCUSSION TITLE
Can I retrieve the version of ICU4X, Unicode, CLDR using C or C++ FFI?

### DISCUSSION THREAD

**Discussion by jlfaucher** [unicode-org/icu4x#4229] [Q&A] [answered]
Hello

Can I retrieve the ICU4X version using C or C++ FFI?
and the Unicode version, the CLDR version?

It would be good to have the Unicode & CLDR version in the release notes.

**Comment by Manishearth** [comment, marked as answer]
Currently this is not possible in the API.

**Reply by jlfaucher** [reply]
Ok, thanks.

**Comment by sffc** [comment]
You can access the CLDR version using deprecated APIs in the icu_testdata crate.

For the ICU4X version, currently you need to look at the Cargo.lock file. You can do this in a build.rs script that you write.

This has now been requested multiple times so I think we should probably support it.

**Reply by sffc** [reply]
To close the loop on this, the versions are now available in the Cargo.toml files which can be queried at build time using the `cargo metadata` tool.

**Reply by sffc** [reply]
See https://github.com/unicode-org/icu4x/pull/4399

This will be shipped in ICU4X 1.5
